src/main/scala/nl.knaw.dans.easy.bagstore/server/BagsServletComponent.scala

```diff
@@ -42,16 +45,26 @@ trait BagsServletComponent extends DebugEnhancedLogging {
         })
     }
 
+
     get("/:uuid") {
       contentType = "text/plain"
       val uuidStr = params("uuid")
       ItemId.fromString(uuidStr)
         .recoverWith {
           case _: IllegalArgumentException => Failure(new IllegalArgumentException(s"invalid UUID string: $uuidStr"))
         }
-        .flatMap(_.toBagId)
-        .flatMap(bagStores.enumFiles(_))
-        .map(bagIds => Ok(bagIds.mkString("\n")))
+        .flatMap{
+          case bagId: BagId =>
+            debug(s"Retrieving item $bagId")
+            request.getHeader("Accept") match {
+              case "application/zip" => bagStores.getStream(bagId, response.outputStream).map(_ => Ok())
+              case "text/plain" | "*/*" | null => bagStores.enumFiles(bagId).map(files => Ok(files.toList.mkString("\n")))
```

Comment:
An empty OK doesn't seem of much use to me, we'll need the stream or its content.